persistence/provider/sql/sqlite/schema.go

```diff
@@ -113,7 +113,7 @@ func createEventStoreSchema(ctx context.Context, db *sql.DB) {
 		`CREATE INDEX repository_query ON event (
 			source_app_key,
 			portable_name,
-			offset,
+			message_id,
```

Comment:
Removing `offset` from this index means that `QueryEvents()` no longer has a good index to use.

There should be a separate index on the table for each variant of the query, and we can name them a little like the driver method they relate to. 

Please see [this comment](https://github.com/dogmatiq/infix/pull/253/files#diff-740d83e6a73bbe31f33fc03cfc0ab57bR360) before changing this, as it might change how many driver methods we have.